### DISCUSSION TITLE
Why can't some parameters be queried in the database?

### DISCUSSION THREAD

**Discussion by shelly-ctrl** [ClickHouse/ClickHouse#58707] [Q&A] [answered]
Why some parameters in clickhouse, such as the database_atomic_delay_before_drop_table_sec parameter, This parameter is not available in the server_settings and settings tables and by using the SHOW STEEINGS command, but it does take effect when I configure it in the config.xml file. Is this parameter not available in the database?

Thanks!!!

**Comment by mlex** [comment, marked as answer]
A setting only shows up in `system.settings` (and related commands like `SHOW SETTINGS`) when the setting is registered in [ServerSettings.h](https://github.com/ClickHouse/ClickHouse/blob/master/src/Core/ServerSettings.h)?

Not all settings are in there though (for historic reasons - some settings existed even before this table was introduced and were never migrated). The `database_atomic_delay_before_drop_table_sec` is one of them. It's not in `ServerSettings.h`. It's only used in [DatabaseCatalog.cpp](https://github.com/ClickHouse/ClickHouse/blob/master/src/Interpreters/DatabaseCatalog.cpp#L185).

This could be an easy PR to earn the ClickHouse contributor badge 😃

**Reply by shelly-ctrl** [reply]
Thank you very much for your answer! Answered my confusion！🥰
